pkg/oauth/apis/oauth/validation/validation_test.go

```diff
@@ -746,3 +746,38 @@ func TestValidateAuthorizeTokensUpdate(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateUserNameField(t *testing.T) {
+	fieldArg := field.NewPath("metadata", "name")
+	valid := "kube:admin"
+	errs := ValidateUserNameField(valid, fieldArg)
+	if len(errs) != 0 {
+		t.Errorf("expected success: %v", errs)
+	}
+
+	errorCases := map[string]struct {
+		name string
+		T    field.ErrorType
+		F    string
+	}{
+		"system:admin": {
+			T: field.ErrorTypeInvalid,
+			F: "metadata.name",
+		},
+	}
+	for k, v := range errorCases {
+		errs := ValidateUserNameField(k, fieldArg)
+		if len(errs) == 0 {
+			t.Errorf("expected failure %s for %v", k, k)
+			continue
+		}
+		for i := range errs {
```

Comment:
is `len(errs) > 1` actually expected?

Reply:
The `expected_err_type` returns an `ErrorList` so I could either implement it like this or add an assert ensuring there's one error. Either way - the end result is the same.